test/e2e/e2e_suite_test.go

```diff
@@ -159,17 +159,23 @@ func loadE2EConfig(configPath string) *clusterctl.E2EConfig {
 	config := clusterctl.LoadE2EConfig(context.TODO(), clusterctl.LoadE2EConfigInput{ConfigPath: configPath})
 	Expect(config).ToNot(BeNil(), "Failed to load E2E config from %s", configPath)
 
-	// Ensuring a CNI file is defined in the config; this will be used at later stages to fill in the CNI_RESOURCES variable in templates
-	Expect(config.Variables).To(HaveKey(clusterctl.CNIPath), "Missing %s variable in the config", clusterctl.CNIPath)
-
 	return config
 }
 
 func createClusterctlLocalRepository(config *clusterctl.E2EConfig, repositoryFolder string) string {
-	clusterctlConfig := clusterctl.CreateRepository(context.TODO(), clusterctl.CreateRepositoryInput{
+	createRepositoryInput := clusterctl.CreateRepositoryInput{
 		E2EConfig:        config,
 		RepositoryFolder: repositoryFolder,
-	})
+	}
+
+	// Ensuring a CNI file is defined in the config and register a FileTransformation to inject the referenced file as in place of the CNI_RESOURCES envSubst variable.
+	Expect(config.Variables).To(HaveKey(CNIPath), "Missing %s variable in the config", CNIPath)
+	cniPAth := config.GetVariable(CNIPath)
+	Expect(cniPAth).To(BeAnExistingFile(), "The %s variable should resolve to an existing file", CNIPath)
+
+	createRepositoryInput.RegisterClusterResourceSetConfigMapTransformation(cniPAth, CNIResources)
```

Comment:
Just to understand, will `createClusterctlLocalRepository ` be called multiple times for adding different CNIs?

Reply:
No, there will be only one call to createClusterctlLocalRepository as of today.
However, inside this func you have to register one ClusterResourceSetConfigMapTransformation for each CNI you want to support.